python/lsst/daf/butler/registry/queries/_builder.py

```diff
@@ -150,6 +154,9 @@ def joinDataset(self, datasetType: DatasetType, collections: Any, *,
             collectionTypes = {CollectionType.RUN}
         else:
             collectionTypes = CollectionType.all()
+            # Do not search calibration collections if requested
+            if self._skipCalibs:
+                collectionTypes -= {CollectionType.CALIBRATION}
```

Comment:
Is this the line that does all the work for this entire PR?

Reply:
Indeed, the rest of it is just passing that flag from registry methods to here.

Reply:
Which raises the question I bring up in Jira. If querying calibration collections never works, why isn't this one line there until it's fixed? I assume there are scenarios where we do successfully query calibration collections through some other path?

Reply:
I want to play with what I suggested on Jira - we could quietly  skip calibration collections when searching in chained ones, but raise an exception if someone searches calibration collection directly. It needs figuring out what collections were requested, may not be very trivial.